relay-server/src/utils/rate_limits.rs

```diff
@@ -372,18 +387,53 @@ where
         let mut enforcement = Enforcement::default();
 
         if let Some(category) = summary.event_category {
-            let event_limits = (self.check)(scoping.item(category), 1)?;
-            let longest = event_limits.longest();
-            enforcement.event = CategoryLimit::new(category, 1, longest);
-
-            // Record the same reason for attachments, if there are any.
-            enforcement.attachments = CategoryLimit::new(
-                DataCategory::Attachment,
-                summary.attachment_quantity,
-                longest,
-            );
-
-            rate_limits.merge(event_limits);
+            if category == DataCategory::Transaction {
+                if summary.transaction_metrics_extracted {
+                    // Check and increment indexing quota.
+                    let index_limits = (self.check)(scoping.item(DataCategory::Transaction), 1)?;
+                    let longest = index_limits.longest();
+                    enforcement.event = CategoryLimit::new(DataCategory::Transaction, 1, longest);
+                    rate_limits.merge(index_limits);
+
+                    // Check processing quota, but DO NOT increment.
+                    let processing_limits =
+                        (self.check)(scoping.item(DataCategory::TransactionProcessed), 0)?;
```

Comment:
Why do we check `TransactionProcessed` quota here? If metrics have already been extracted, the transaction has been processed. One reason is that this function is called again later in the pipeline, but in that case we shouldn't increment the limit below?

Reply:
This branch is **if** metrics were already extracted, i.e. the 2nd time in the pipeline (or the first time if an earlier relay extracted the metrics).  The reason for this check is that we do not want to index a transaction which we're not also extracting metrics from.  So even though we might still have indexing quota we should not index this if we're out of processing quota.

But writing this makes me realise this code is incorrect.  If this quota is exhausted we should **also** mark the event as enforced.  This it doesn't currently do.